### PR TITLE
Copy / paste support for images

### LINKED ISSUE #108: Copy / Paste support for lists.
Implement copy / paste support for lists.

### PR DESCRIPTION
**Pull request by diegoreymendez**
Closes #108.

**Description:**

- Implements copy / paste support for images.
- Reduces the visibility for some methods that were `public`.
- Improves the attachment code a bit, and triggers updates only on URL changes.
- Standardizes delegate-type protocols to follow Apple's conventions for naming.

**How to test:**

Test 1:

1. Copy an image from another app (a good source is Pages.app).
2. Paste it in Aztec.

Test 2: Insert an image regularly
Test 3: Insert an image in HTML mode.

For all tests verify the following:

- Make sure the image is properly added to the post.
- Make sure the HTML is updated properly.
- Make sure the layout isn't broken.

### CODE REVIEW

File: Aztec/Classes/Public/TextKit/TextAttachment.swift [hunk 1]
```diff
@@ -130,38 +138,48 @@ public class TextAttachment: NSTextAttachment
         return CGRect(origin: CGPointZero, size: CGSize(width: width, height: onScreenHeight(width)))
     }
 
-    func updateImage() {
+    func updateImage(inTextContainer textContainer: NSTextContainer? = nil) {
```
Comment (SergioEstevao) [inline Aztec/Classes/Public/TextKit/TextAttachment.swift]: Loved this change, avoiding the textContainer variable on the class!

File: Aztec/Classes/Public/TextKit/TextAttachment.swift [hunk 2]
```diff
@@ -1,20 +1,22 @@
 import Foundation
 import UIKit
 
-public protocol TextAttachmentImageProvider {
-    func image(forURL url: NSURL, forAttachment attachment: TextAttachment, onSuccess success: (UIImage) -> (), onFailure failure: () -> ()) -> UIImage?
+protocol TextAttachmentImageProvider {
+    func textAttachment(textAttachment: TextAttachment, imageForURL url: NSURL, onSuccess success: (UIImage) -> (), onFailure failure: () -> ()) -> UIImage
```
Comment (diegoreymendez) [inline Aztec/Classes/Public/TextKit/TextAttachment.swift]: Even though we don't have a specific rule in our style guide, [it does mention that RayWenderlich's guide is a good starting point for reference](https://github.com/wordpress-mobile/swift-style-guide).

You can see in [this issue](https://github.com/raywenderlich/swift-style-guide/issues/175) that they're already working towards adding this convention to their docs.  Here's [one of the commits](https://github.com/fabienwarniez/swift-style-guide/commit/25ccd3935c20c0954ff97d1688a66386aa5f6191) showing some clear examples.

Additionally, there's no indication that this only pertains to Objective-C code in Apple's delegates (UITableViewDelegate, UIPickerViewDelegate, etc).

All in all, going in this direction would basically ensure we'd be following a widely used convention.

**Comment by SergioEstevao** [comment]
:shipit:

**Comment by diegoreymendez** [comment]
Ty Sergio!

**Pull request merged**